diffrax/_solver/foster_langevin_srk.py

```diff
@@ -0,0 +1,410 @@
+import abc
+from collections.abc import Callable
+from typing import Any, Generic, TypeVar
+
+import equinox as eqx
+import equinox.internal as eqxi
+import jax
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.tree_util as jtu
+from equinox import AbstractVar
+from jax import vmap
+from jaxtyping import PyTree
+
+from .._custom_types import (
+    AbstractBrownianIncrement,
+    BoolScalarLike,
+    DenseInfo,
+    RealScalarLike,
+)
+from .._local_interpolation import LocalLinearInterpolation
+from .._solution import RESULTS
+from .._term import (
+    AbstractTerm,
+    MultiTerm,
+    UnderdampedLangevinDiffusionTerm,
+    UnderdampedLangevinDriftTerm,
+    UnderdampedLangevinLeaf,
+    UnderdampedLangevinTuple,
+    UnderdampedLangevinX,
+    WrapTerm,
+)
+from .base import AbstractItoSolver, AbstractStratonovichSolver
+
+
+_ErrorEstimate = TypeVar("_ErrorEstimate", None, UnderdampedLangevinTuple)
+ULDArgs = tuple[
+    UnderdampedLangevinX,
+    UnderdampedLangevinX,
+    Callable[[UnderdampedLangevinX], UnderdampedLangevinX],
+]
+
+
+def _get_args_from_terms(
+    terms: MultiTerm[tuple[AbstractTerm[Any, RealScalarLike], AbstractTerm]],
+) -> tuple[PyTree, PyTree, Callable[[UnderdampedLangevinX], UnderdampedLangevinX]]:
+    drift, diffusion = terms.terms
+    if isinstance(drift, WrapTerm):
+        assert isinstance(diffusion, WrapTerm)
+        drift = drift.term
+        diffusion = diffusion.term
+
+    assert isinstance(drift, UnderdampedLangevinDriftTerm)
+    assert isinstance(diffusion, UnderdampedLangevinDiffusionTerm)
+    gamma = drift.gamma
+    u = drift.u
+    f = drift.grad_f
+    return gamma, u, f
+
+
+def _broadcast_pytree(source, target_tree):
+    # Broadcasts the source PyTree to the shape and PyTree structure of
+    # target_tree_shape. Requires that source is a prefix tree of target_tree
+    # This is used to case gamma and u to the shape of x0 and v0
```

Comment:
'case' -> 'broadcast'?

Reply:
I think I wanted to write "cast" but made a typo